metricproviders/wavefront/wavefront.go

```diff
@@ -0,0 +1,240 @@
+	p.logCtx.Warn("Wavefront provider should not execute the Terminate method")
+	return measurement
+}
+
+// GarbageCollect is a no-op for the Wavefront provider
+func (p *Provider) GarbageCollect(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, limit int) error {
+	return nil
+}
+
+func (p *Provider) evaluateResult(result interface{}, metric v1alpha1.Metric) v1alpha1.AnalysisPhase {
+	successCondition := false
+	failCondition := false
+	var err error
+
+	if metric.SuccessCondition != "" {
+		successCondition, err = evaluate.EvalCondition(result, metric.SuccessCondition)
+		if err != nil {
+			p.logCtx.Warning(err.Error())
+			return v1alpha1.AnalysisPhaseError
+		}
+	}
+	if metric.FailureCondition != "" {
+		failCondition, err = evaluate.EvalCondition(result, metric.FailureCondition)
+		if err != nil {
+			return v1alpha1.AnalysisPhaseError
+		}
+	}
+
+	switch {
+	case metric.SuccessCondition == "" && metric.FailureCondition == "":
+		//Always return success unless there is an error
+		return v1alpha1.AnalysisPhaseSuccessful
+	case metric.SuccessCondition != "" && metric.FailureCondition == "":
+		// Without a failure condition, a measurement is considered a failure if the measurement's success condition is not true
+		failCondition = !successCondition
+	case metric.SuccessCondition == "" && metric.FailureCondition != "":
+		// Without a success condition, a measurement is considered a successful if the measurement's failure condition is not true
+		successCondition = !failCondition
+	}
+
+	if failCondition {
+		return v1alpha1.AnalysisPhaseFailed
+	}
+
+	if !failCondition && !successCondition {
+		return v1alpha1.AnalysisPhaseInconclusive
+	}
+
+	// If we reach this code path, failCondition is false and successCondition is true
+	return v1alpha1.AnalysisPhaseSuccessful
+}
+
+func (p *Provider) processResponse(metric v1alpha1.Metric, response *wavefront_api.QueryResponse) (string, v1alpha1.AnalysisPhase, error) {
+
+	if len(response.TimeSeries) == 1 {
+		series := response.TimeSeries[0]
+		result := series.DataPoints[0][1]
+		if math.IsNaN(result) {
+			return fmt.Sprintf("%.2f", result), v1alpha1.AnalysisPhaseInconclusive, nil
+		}
+		newStatus := p.evaluateResult(result, metric)
+		return fmt.Sprintf("%.2f", result), newStatus, nil
+
+	} else if len(response.TimeSeries) > 1 {
+		results := make([]float64, 0, len(response.TimeSeries))
+		valueStr := "["
+		for _, series := range response.TimeSeries {
+			value := series.DataPoints[0][1]
+			valueStr = valueStr + fmt.Sprintf("%.2f", value) + ","
+			results = append(results, value)
+		}
+		if len(valueStr) > 1 {
+			valueStr = valueStr[:len(valueStr)-1]
+		}
+		valueStr = valueStr + "]"
+		for _, result := range results {
+			if math.IsNaN(result) {
+				return valueStr, v1alpha1.AnalysisPhaseInconclusive, nil
+			}
+		}
+		newStatus := p.evaluateResult(results, metric)
+		return valueStr, newStatus, nil
+
+	} else {
+		return "", v1alpha1.AnalysisPhaseFailed, nil
```

Comment:
Can you write a quick unit test where Wavefront would return a result that has no TimeSeries?